source/fundamentals/crud/write-operations/upsert.txt

```diff
@@ -4,8 +4,200 @@ Insert or Update in a Single Operation
+
+To update all colors, query all the documents in the ``paint_inventory``
+collection, set the update to increase ``qty`` by ``20`` and 
+pass the query to the ``updateMany()`` method: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/InsertUpdate.java
+   :language: java
+   :dedent:
+   :start-after: begin updateManyExample
+   :end-before: end updateManyExample
+
+The method returns:
+
+.. code-block:: json   
+
+    AcknowledgedUpdateResult{ matchedCount=9, modifiedCount=9, upsertedId=null } 
+
+The following shows the documents in the ``paint_inventory`` collection:
+
+.. code-block:: json   
+
+    { "_id": 1, "color": "red", "qty": 25 }
+    { "_id": 2, "color": "purple", "qty": 28 }
+    { "_id": 3, "color": "blue", "qty": 20 }
+    { "_id": 4, "color": "white", "qty": 20 }
+    { "_id": 5, "color": "yellow", "qty": 26 }
+    { "_id": 6, "color": "pink", "qty": 20 }
+    { "_id": 7, "color": "green", "qty": 20 }
+    { "_id": 8, "color": "black", "qty": 28 }
+    { "_id": { "$oid": "606787841601f9443bc291ce" }, "color": "orange", "qty": 30 }
+
+``replaceOne()``
+----------------
+
+The replaceOne() method replaces a document that matches the query
+filter. If the query filter matches more than one document, the method
+will replace the first occurrence of a match in the collection.
+
+Example
+~~~~~~~
+
+From the previous example, we noticed orange got incremented by
+accident and we need to replace the orange document to have ten cans of
+paint. 
+
+To replace the orange document, query the ``paint_inventory`` collection
+where the ``color`` is ``"orange"``, create a document where the
+``color`` is ``orange`` and the ``qty`` is ``10`` and pass the query to the
+``replaceOne()`` method:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/InsertUpdate.java
+   :language: java
+   :dedent:
+   :start-after: begin updateManyExample
+   :end-before: end updateManyExample
+
+The method returns:
+
+.. code-block:: json   
+
+    AcknowledgedUpdateResult{ matchedCount=1, modifiedCount=1, upsertedId=null } 
+
+The following shows the documents in the ``paint_inventory`` collection:
+
+.. code-block:: json   
+
+    { "_id": 1, "color": "red", "qty": 25 }
+    { "_id": 2, "color": "purple", "qty": 28 }
+    { "_id": 3, "color": "blue", "qty": 20 }
+    { "_id": 4, "color": "white", "qty": 20 }
+    { "_id": 5, "color": "yellow", "qty": 26 }
+    { "_id": 6, "color": "pink", "qty": 20 }
+    { "_id": 7, "color": "green", "qty": 20 }
+    { "_id": 8, "color": "black", "qty": 28 }
+    { "_id": { "$oid": "60678b161601f9443bc31df6" }, "color": "orange", "qty": 10 }
+    
+See the following documentation for more information about the updating
+a document:
+
+- :manual:`db.collection.updateOne() </reference/method/db.collection.updateOne/>`
+- :manual:`db.collection.updateMany() </reference/method/db.collection.updateMany/>`
+- :manual:`db.collection.replaceOne() </reference/method/db.collection.replaceOne/>`
```

Comment:
question: I'm unsure what this content has to do with upserting? Might be better to remove it to focus on the main topic.